### DISCUSSION TITLE
`TypeError: Factory, Object, or Array expected` when custom bundling

### DISCUSSION THREAD

**Discussion by twiertzema** [josdejong/mathjs#2668] [Q&A] [answered]
When attempting to pull in only the parts of the library that I need (following the examples from https://mathjs.org/docs/custom_bundling.html), I receive the following error:
```
Uncaught TypeError: Factory, Object, or Array expected
    at flattenImports (import.js:102)
    at import.js:80
    at Array.forEach (<anonymous>)
    at flattenImports (import.js:80)
    at Object.mathImport [as import] (import.js:108)
    at create (create.js:235)
```
This is my importing code:
```js
import { create, matrix, multiply } from 'mathjs'
const math = create({ matrix, multiply })
```

I am using version `6.0.2`.

**Comment by josdejong** [comment, marked as answer]
The `create` function expects factory functions, like `matrixDependencies` and `multiplyDependencies`. The docs page you're referring to shows an example: https://mathjs.org/docs/custom_bundling.html
